notify/templates.go

```diff
@@ -0,0 +1,240 @@
+package notify
+
+import (
+	"context"
+	"fmt"
+	tmplhtml "html/template"
+	"net/url"
+	"path/filepath"
+	tmpltext "text/template"
+	"text/template/parse"
+
+	"github.com/grafana/alerting/templates"
+	v2 "github.com/prometheus/alertmanager/api/v2"
+	"github.com/prometheus/alertmanager/notify"
+	"github.com/prometheus/alertmanager/template"
+	"github.com/prometheus/common/model"
+)
+
+type TestTemplatesConfigBodyParams struct {
+	// Alerts to use as data when testing the template.
+	Alerts []*PostableAlert `json:"alerts"`
+
+	// Template string to test.
+	Template string `json:"template"`
+
+	// Name of the template file.
```

Comment:
I was able to make it work without having this field required, but if we want to keep it we need to tell Sonia and Vika about it because it's not in the design doc.

Reply:
OK please just let Sonia and Vika know about it and also update the design doc 🙂

Reply:
On further thought I think I want to disagree with adding `name`, or at least making it a required field.

The reason the design doc doesn't have a `name` field in the request JSON is because from a UX perspective I don't want users to have to choose a name in the UI as a pre-requisite to previewing templates. I don't think it's required for it to work and so all it does is create extra work for the user just to get a simple preview.

If and when the user is satisfied with the preview can they choose a name in the UI and click the Save button.

Reply:
Ah, I see what you're saying. I agree:

If a user wants to their test to be tested as if it replaces the currently existing template of the same name, it is required. Otherwise, if they're creating a new one and it's blank it will work fine without it in the current code.

Reply:
Fixed some edge cases with empty name and added tests, should have better support now.